detox/src/Detox.test.js

```diff
@@ -95,12 +96,21 @@ describe('Detox', () => {
     expect(() => detox.cleanup()).not.toThrowError();
   });
 
+  it(`Calling detox.init() twice returns the same promise`, async () => {
+    Detox = require('./Detox');
+    detox = new Detox({deviceConfig: validDeviceConfig});
+    const promise1 = await detox.init();
+    const promise2 = await detox.init();
+    expect(promise1).toBe(promise2);
+  });
+
   it(`Not passing --cleanup should keep the currently running device up`, async () => {
     Detox = require('./Detox');
     detox = new Detox({deviceConfig: validDeviceConfig});
     await detox.init();
+    const device = detox.device;
     await detox.cleanup();
-    expect(detox.device.shutdown).toHaveBeenCalledTimes(0);
+    expect(device.shutdown).toHaveBeenCalledTimes(0);
   });
 
```

Comment:
The additional coverage is good, but I kind'a wish you'd cover the multiple async init rejection techniques.

It would make it both tested and also better documented (for example, if you specify explicitly that before-each should throw if init failed due to a _timeout_, etc.)